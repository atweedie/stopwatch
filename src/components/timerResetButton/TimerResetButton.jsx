import { useContext } from "react";
import { TimerState } from "../../modules/timerStateContext";
import {TIMER_TICK_DURATION_IN_MS} from "../../constants";

const TimerResetButton = () => {
    const {setElapsedTime, setIsRunning} = useContext(TimerState);

    const resetTimer = () => {
        setIsRunning(false);

        setTimeout(() => {setElapsedTime(0)}, TIMER_TICK_DURATION_IN_MS)
    }

    return (
        <button onClick={resetTimer}>
            Reset
        </button>
    )
}

export default TimerResetButton