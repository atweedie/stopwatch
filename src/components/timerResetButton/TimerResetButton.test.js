import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react';
import TimerResetButton from './TimerResetButton';

const stubSetElapsedTime = jest.fn();
const stubSetIsRunning = jest.fn();
const stubUseContext = jest.fn();

const defaultContext = {
    isRunning: false,
    setElapsedTime: stubSetElapsedTime,
    setIsRunning: stubSetIsRunning
}

const renderWithContext = (context = defaultContext) => {
    React.useContext = stubUseContext.mockReturnValue(context)
    return render(<TimerResetButton/>)
};

beforeEach(() => {
    jest.useFakeTimers();
})

afterEach(() => {
    jest.useRealTimers();
});

test('should render as expected', () => {
const component = renderWithContext();
    
    expect(component.container).toMatchSnapshot();
});

test('should call setElapsedTime with expected parameters on button click', () => {
    renderWithContext();

    const button = screen.getByText('Reset');
        
    fireEvent.click(button)

    jest.runAllTimers();

    expect(stubSetElapsedTime).toHaveBeenCalledWith(0);
});

test('should call setIsRunning with expected parameters on button click', () => {
    renderWithContext();

    const button = screen.getByText('Reset');
        
    fireEvent.click(button)

    jest.runAllTimers();

    expect(stubSetIsRunning).toHaveBeenCalledWith(false);
});