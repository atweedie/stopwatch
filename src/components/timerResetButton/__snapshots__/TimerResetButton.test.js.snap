// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <button
    class="button"
    id="stopwatch-reset"
  >
    Reset
  </button>
</div>
`;
